src/helpers/getByAPI.js

```diff
@@ -44,6 +44,14 @@ const getChildrenAsText = (children, TextComponent, textContent = []) => {
     }
 
     if (child?.props?.children) {
+      // Bail on traversing text children down the tree if current node (child)
+      // has no text. In such situations, react-test-renderer will traverse down
+      // this tree in a separate call and run this query again. As a result, the
+      // query will match a deeper text node that may have text some content.
```

Comment:
```suggestion
      // query will match a deeper text node that has the same text content.
```

Reply:
My intention was "my have some text content" because it may happen it's still empty. Make sense to update the text this way?

Reply:
"some" or "same" text content?

Reply:
"some". Like:
```jsx
<Text> 
  // no text
  <Text>
    // no text here as well
    <Text>some text content</Text>
  </Text>
</Text>
```

Reply:
Maybe we can go with: "As a result, ByText queries will match a deepest text node that matches requested text.", wdyt?

Reply:
Sounds good 👍 "a deepest" -> "the deepest", rest is fine